lib/semantics/resolve-names.cc

```diff
@@ -65,7 +66,7 @@ class ImplicitRules {
   void set_isImplicitNoneExternal(bool x) { isImplicitNoneExternal_ = x; }
   void set_inheritFromParent(bool x) { inheritFromParent_ = x; }
   // Get the implicit type for identifiers starting with ch. May be null.
-  std::optional<const DeclTypeSpec> GetType(char ch) const;
+  const DeclTypeSpec *GetType(char ch) const;
```

Comment:
If the function's result cannot be a null pointer, please make its return type a reference.

Reply:
It can be null. The change was to return the DeclTypeSpec by reference rather than by value, but it's still optional.